### PR TITLE
Settings for v1.3 production runs

### PR DESCRIPTION
**Pull request by lea-hagen**
Here's a starting point for the upcoming round of production runs (SMIDGE, METAL, PHAT).  Let's keep the discussion in #322 (not here) so it's all in one place.

Relevant parameters/settings:
* AST selection
  * 40 flux bins, 50 SEDs per bin
  * bins covering the whole range of the model grid, not just the catalog
  * each SED is *ONLY* replicated across source density bins
* noise model: toothpick (not truncheon)
* grid parameters: [min, max, step_size]
  * log_age = [6.0, 10.13, 0.15]
  * Av = [0.01, 10.055, 0.1], flat prior
  * Rv = [2.0, 6.0, 0.5], flat prior
  * fA = [0.0, 1.0, 0.25], flat prior

Things that need to be decided, in addition to confirming what's above:
* extinction law
* what metallicity list to use for each galaxy
* perhaps others?

### CODE REVIEW

**Comment by karllark** [comment]
This pull request **introduces 9 alerts** when merging 6de7a1a56263b7198310e9613cb9af38a664bdb0 into 974fb72f3a1974d25d3a36304122464c6e86b68c - [view on LGTM.com](https://lgtm.com/projects/g/BEAST-Fitting/beast/rev/pr-033d4af800b023c4f946488c45cae2ada01de5bb)

**new alerts:**

* 5 for Unused local variable
* 4 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by lea-hagen** [comment]
Current grid info:
* log_age = [6.0, 10.13, 0.1]
* Av = [0.01, 10.0, 0.05]
* Rv = [2.0, 6.0, 0.5]
* fA = [0.0, 1.0, 0.2]
* log Z = [-2.3, -2.0, -1.7, -1.4, -1.1, -0.8, -0.5, -0.2,  0.1]

The last thing to do is finalize the extinction model!

**Comment by karllark** [comment]
This pull request **introduces 6 alerts** when merging 9b6d82e6b7cea3bc3498da35e1f1aa78838573ea into 97db592ba847d18294a1761a206ad0fc09fd2274 - [view on LGTM.com](https://lgtm.com/projects/g/BEAST-Fitting/beast/rev/pr-43fffa3443a4eee119f3400894a3f7c929cf886b)

**new alerts:**

* 5 for Unused local variable
* 1 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: beast/examples/production_runs_2019/datamodel.py [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+# Alternative: MIST -- v1, no rotation
+#oiso = isochrone.MISTWeb()
+
+# Stellar Atmospheres library definition
+osl = stellib.Tlusty() + stellib.Kurucz()
+
+################
+
+### Dust extinction grid definition
+extLaw = extinction.Gordon16_RvFALaw()
```
Comment (lcjohnso) [inline beast/examples/production_runs_2019/datamodel.py]: I recommend using new dust_extinction package to enable revised RvFA extinction curve model where ALaw = F20 (Fitzpatrick et al. 2020, in prep) and BLaw = Gordon03_SMCBar (Gordon et al. 2003):
`extLaw = extinction.Generalized_RvFALaw(ALaw=extinction.Generalized_DustExt(curve='F20'))`

Comment (lcjohnso) [reply beast/examples/production_runs_2019/datamodel.py]: Writing the choices for both ALaw and BLaw explicitly:
`extLaw = extinction.Generalized_RvFALaw(ALaw=extinction.Generalized_DustExt(curve='F20'), BLaw=extinction.Gordon03_SMCBar())`

**Comment by karllark** [comment]
This pull request **introduces 7 alerts** when merging 8dd1a1f8b4442a37b7180ac124db3a023b961963 into 97db592ba847d18294a1761a206ad0fc09fd2274 - [view on LGTM.com](https://lgtm.com/projects/g/BEAST-Fitting/beast/rev/pr-406a4e9df595f2986d63278dd1cec4c028d73178)

**new alerts:**

* 5 for Unused local variable
* 1 for Unused import
* 1 for Variable defined multiple times

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by lea-hagen** [comment]
`datamodel.py` is ready to go.

We need to implement subgridding in order to accommodate the large model grid.  I'm in the processes of updating `run_beast_production.py`. Once I finish that, this pull request will be ready to be merged.

**Comment by karllark** [comment]
Is this pull request ready for merging?

**Comment by lea-hagen** [comment]
Any changes at this point will be minor.  I think this can be merged, unless we want to wait for #364 to be okayed.

**Comment by karllark** [comment]
This pull request **introduces 21 alerts** when merging a631e3c11e599ac5d7dd665f7fa51ce2b4045ae8 into de50ca4d05ba681fd6e2450fc12841d728c9eb23 - [view on LGTM.com](https://lgtm.com/projects/g/BEAST-Fitting/beast/rev/pr-2d7fd32b3e22c9a45a8e6932f6f4548aa1ee29e4)

**new alerts:**

* 9 for Unused import
* 5 for Unreachable code
* 3 for Unused local variable
* 2 for Variable defined multiple times
* 1 for Wrong name for an argument in a call
* 1 for Syntax error

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by karllark** [comment]
Lots of LGTM issues with the latest commit.  Did you want to clean some of them up?

**Comment by lea-hagen** [comment]
Yes, I'll do that!  I guess I was a bit too enthusiastic on a Friday afternoon and missed the notification about those :)

**Comment by lea-hagen** [comment]
Notes on the LGTM suggestions (listed here for future reference):
* Unreachable code: there are `if` statements for calling code depending on whether or not subgrids are in use, which would be fixed by putting that setting in `datamodel`.  I think I'll do that.
* Wrong name for argument in code: will be fixed when #364 is merged
* Syntax error: something with Python 2.7 (can LGTM be told to use python > 3?)

**Comment by karllark** [comment]
This pull request **introduces 7 alerts** when merging 59130511ddbde3b9585f470423ec82423ccde99c into de50ca4d05ba681fd6e2450fc12841d728c9eb23 - [view on LGTM.com](https://lgtm.com/projects/g/BEAST-Fitting/beast/rev/pr-62a6cd41aba95994cf2c721dd0f6daf5d069f4e2)

**new alerts:**

* 4 for Unreachable code
* 1 for Wrong name for an argument in a call
* 1 for Syntax error
* 1 for Variable defined multiple times

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by karllark** [comment]
Is this now ready for merge?  If so, remove the WIP from the title and I'll do the merge.

**Pull request merged**